cli/dts/lib.deno.unstable.d.ts

```diff
@@ -1264,85 +1258,90 @@ declare namespace Deno {
+   * You can change the address to listen on using the `hostname` and `port`
+   * options. The below example serves on port 3000.
    *
    * ```ts
-   * Deno.serve({
-   *   fetch: (_req) => new Response("Hello, world"),
-   *   port: 3000
-   * });
+   * Deno.serve({ port: 3000 }, (_req) => new Response("Hello, world"));
    * ```
    *
-   * You can close the server by passing a `signal` option. To wait for the server
-   * to close, await the promise returned from the `Deno.serve` API.
+   * You can stop the server with an AbortSignal. The abort signal needs to be
+   * passed as the `signal` option in the options bag. The server aborts when
+   * the abort signal is aborted. To wait for the server to close, await the
+   * promise returned from the `Deno.serve` API.
    *
    * ```ts
    * const ac = new AbortController();
    *
-   * Deno.serve({
-   *   fetch: (_req) => new Response("Hello, world"),
-   *   signal: ac.signal
-   * }).then(() => {
-   *   console.log("Server closed");
-   * });
+   * Deno.serve({ signal: ac.signal }, (_req) => new Response("Hello, world"))
+   *  .then(() => console.log("Server closed"));
    *
    * console.log("Closing server...");
    * ac.abort();
    * ```
    *
-   * `Deno.serve` function prints the message `Listening on http://<hostname>:<port>/`
-   * on start-up by default. If you like to change this message, you can specify
-   * `onListen` option to override it.
+   * By default `Deno.serve` prints the message `Listening on http://<hostname>:<port>/`
+   * on start up. If you like to change this behaviour, you can specify a custom
+   * `onListen` callback.
    *
    * ```ts
    * Deno.serve({
-   *   fetch: (_req) => new Response("Hello, world"),
    *   onListen({ port, hostname }) {
    *     console.log(`Server started at http://${hostname}:${port}`);
    *     // ... more info specific to your server ..
    *   },
+   *   handler: (_req) => new Response("Hello, world"),
    * });
    * ```
    *
-   * To enable TLS you must specify `key` and `cert` options.
+   * To enable TLS you must specify the `key` and `cert` options.
    *
    * ```ts
    * const cert = "-----BEGIN CERTIFICATE-----\n...\n-----END CERTIFICATE-----\n";
    * const key = "-----BEGIN PRIVATE KEY-----\n...\n-----END PRIVATE KEY-----\n";
-   * Deno.serve({
-   *   fetch: (_req) => new Response("Hello, world"),
-   *   cert,
-   *   key
-   * });
-   *
-   * @param options The options. See `ServeInit` and `ServeTlsInit` documentation for details.
+   * Deno.serve({ cert, key }, (_req) => new Response("Hello, world"));
    *
    * @category HTTP Server
    */
   export function serve(
-    options?: ServeInit | ServeTlsInit,
+    handler: ServeHandler,
+    options?: ServeOptions | ServeTlsOptions,
+  ): Promise<void>;
+  export function serve(
+    options: ServeOptions | ServeTlsOptions,
+    handler: ServeHandler,
+  ): Promise<void>;
+  export function serve(
+    options: ServeInit & (ServeOptions | ServeTlsOptions),
```

Comment:
Do we need all three? This is just to provide easier DX - but it ends up complicating things because people have to understand there are multiple signatures.

Let's just choose one...